### ISSUE TITLE
Problem with devise_for method in the route file. Completed 500 Internal Server Error in fatal - exception reentered:  

### ISSUE THREAD

**Issue by mayordwells** [plataformatec/devise#5052]
## Environment

- Ruby **[2.6.2]**
- Rails **[6.0.0.beta3]**
- Devise **[4.6.2]**

## Current behavior
Problem with `devise_for` method in the route file
`devise_for :users, path_prefix: 'auth'`
I also have this application_controller
```
before_action :configure_permitted_parameters, if: :devise_controller?
before_action :authenticate_user!
```
It goes into an infinite loop, the server doesn't shut down with ctrl + c, and I always had to kill the process.
<img width="949" alt="Screen Shot 2019-03-27 at 9 21 51 PM" src="https://user-images.githubusercontent.com/1370340/55101979-70f77600-50d6-11e9-812b-c512fe72527d.png">

When I comment out that line in the route file, i'm able to access some other pages. 

The same code works well for Rails 5.2.2 and Ruby 2.6.1. But when I try with Rails 5.2.2.1 and Ruby 2.6.2, I got this error. Now it also doesn't work with Rails 6.0.0.beta3 and Ruby 2.6.2

When this line is commented out `# devise_for :users, path_prefix: 'auth'` I get the following error.

```
=> Booting Puma
=> Rails 6.0.0.beta3 application starting in development 
=> Run `rails server --help` for more startup options
DEPRECATION WARNING: Single arity template handlers are deprecated.  Template handlers must
now accept two parameters, the view object and the source for the view object.
Change:
  >> Class#call(template)
To:
  >> Class#call(template, source)
 (called from <main> at /Users/Mayor/Workspace/welds/config/environment.rb:5)
DEPRECATION WARNING: Single arity template handlers are deprecated.  Template handlers must
now accept two parameters, the view object and the source for the view object.
Change:
  >> Class#call(template)
To:
  >> Class#call(template, source)
 (called from <main> at /Users/Mayor/Workspace/welds/config/environment.rb:5)
Puma starting in single mode...
* Version 3.12.1 (ruby 2.6.2-p47), codename: Llamas in Pajamas
* Min threads: 1, max threads: 1
* Environment: development
* Listening on tcp://localhost:3000
Use Ctrl-C to stop
Started GET "/" for ::1 at 2019-03-27 23:26:56 +0300
   (0.8ms)  SELECT "schema_migrations"."version" FROM "schema_migrations" ORDER BY "schema_migrations"."version" ASC
Processing by HomeController#home_market as HTML
Completed 500 Internal Server Error in 430ms (ActiveRecord: 0.0ms | Allocations: 113508)



NoMethodError - undefined method `authenticate_user!' for #<HomeController:0x00007fbdbe3f2e50>
Did you mean?  authenticate_with_http_digest:

Started POST "/__better_errors/bad57dd0ddbe717c/variables" for ::1 at 2019-03-27 23:26:58 +0300
``` 

Unfortunately, the repo is private, I can only add a user into a forked version of it, I can't make it public.

## Expected behavior
Expected the app to work as normal as it worked in Rails 5.2.2 and Ruby 2.6.1

**Comment by feliperenan** [comment]
Hi @mayordwells, thanks for the issue report.

Could you please provide us a sample application that reproduces the issue in isolation? 

That would help us find the issue.

Thanks

**Comment by mayordwells** [comment]
Hi @feliperenan thanks for your response. I have just added you to a [repo](https://github.com/mayordwells/welds/invitations). You can test the 2 scenarios reported above in there. `dockerize-app` branch and `upgrade-to-rails-6`. The former works fine on Rails 5.2.2.

**Comment by mracos** [comment]
@mayordwells, I think it will be better if you provide us a sample application that we can download and test locally. So all of us can try to help you with this issue. 😄

**Comment by mayordwells** [comment]
@mracos I'm sorry, Problem is the application is not mine and I'd have to seek permission to make it public. Which I'm not sure I can. I can also add you to the repo if that is necessary?

**Comment by mracos** [comment]
@mayordwells Instead of trying to make your application public, try to do an application from scratch (think of a `rails new && rails g devise:install` plus code modifications) that reproduces the error you are facing.

It's better because we can debug the issue in isolation and everyone can chime in and help.

**Comment by mayordwells** [comment]
@mracos okay i'll do that

**Comment by atstockland** [comment]
I have a seemingly similar situation immediately after upgrading from 4.6.1 to 4.6.2.  Downgrading back to 4.6.1 clears this error. 

NameError - uninitialized constant User:
  config/routes.rb:176:in `block in <main>'
  config/routes.rb:1:in `<main>'

2019-03-31 20:03:24 -0600: Read error: #<NameError: uninitialized constant ActionText::Engine::ApplicationController>

Line 176 in my routes file is devise_for :users, skip: [:registrations]

**Comment by mayordwells** [comment]
Oh great, now that I have someone to back me up on this, just tried to downgrade to 4.6.1 and 4.6.0 now, `devise_for` still didn't work with rails 6.0.0.beta3 and ruby 2.6.2. 

Debugging further, I created a fresh project and installed only devise with it, I noticed that 6.0.0.beta3 and ruby 2.6.2 and devise 4.6.2 worked perfectly well on a fresh project. 

But on copying the Gemfile from my project to the fresh project, it stopped working. 

This bug is introduced by one of the Gems in our project @atstockland 

You can test it here - https://github.com/mayordwells/test-devise-for/commits/master

**Comment by mracos** [comment]
@mayordwells Given that the problem does not see to be with devise itself, I will close this issue then. 
If you find out that the problem is indeed with devise feel free to reopen it.
Thanks!

**Comment by atstockland** [comment]
“just tried to downgrade to 4.6.1 and 4.6.0 now, devise_for still didn't work with rails 6.0.0.beta3 and ruby 2.6.2.”

Did you verify in gemfile.lock that devise actually downgraded?  Simply changing:

     gem 'devise', '~> 4.6', '>= 4.6.2'
      to
      gem 'devise', '~> 4.6', '>= 4.6.1’

will not downgrade devise.  In my haste, I made that mistake.  Use gem 'devise', '4.6.1’ to downgrade.  

In my debugging efforts I have found that 4.6.1 and beta3 definitely work fine.  I have been digging around in Devise source...but, not sure what could be causing my issue. For now, I just settled on 4.6.1.
